csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs

```diff
@@ -118,6 +118,16 @@ public void AppendExecutionProvider_CUDA(int deviceId)
             NativeApiStatus.VerifySuccess(NativeMethods.OrtSessionOptionsAppendExecutionProvider_CUDA(handle, deviceId));
         }
 
+        /// <summary>
+        /// Append a CUDA EP instance (based on specified configuration) to the SessionOptions instance
+        /// Use only if you have the onnxruntime package specific to this Execution Provider.
+        /// </summary>
+        /// <param name="cudaProviderOptions">CUDA EP provider options to configure the CUDA EP instance</param>
+        public void AppendExecutionProvider_CUDA(OrtCUDAProviderOptions cudaProviderOptions)
+        {
+            NativeApiStatus.VerifySuccess(NativeMethods.SessionOptionsAppendExecutionProvider_CUDA(handle, cudaProviderOptions.Handle));
```

Comment:
Main interface being made available to a user - configure and append a CUDA EP to SessionOptions